### PR TITLE
Configurable native faucet mode

### PR DESCRIPTION
**Pull request by grafnu**

### CODE REVIEW

File: cmd/faucet [hunk 1]
```diff
@@ -53,38 +62,97 @@ if [ -n "$1" ]; then
...
+    export FAUCET_EXCEPTION_LOG=$RUNTIME/faucet_exception.log
+    export FAUCET_CONFIG_STAT_RELOAD=1
+    export FAUCET_LISTEN_PORT=$ext_ofpt
+    export FAUCET_PROMETHEUS_PORT=${prom_pt:-$PROM_TARGET}
+    export FAUCET_CONFIG=$RUNTIME/faucet.yaml
+    export FAUCET_EVENT_SOCK=$RUNTIME/$FAUCET_SOCK
+    export WSAPI_LISTEN_PORT=4567
+    echo Launching faucet $FAUCET_LISTEN_PORT $FAUCET_PROMETHEUS_PORT $WSAPI_LISTEN_PORT
+    [ -f $RYU_CONF ] || (echo $RYU_CONF not found.; exit 1)
+    faucet --ryu-config-file=$RYU_CONF --ryu-ofp-tcp-listen-port=$FAUCET_LISTEN_PORT &
```
Comment (henry54809) [inline cmd/faucet:109]: Where does it start gauge natively?

**Comment by grafnu** [comment]
Updated with gauge, PTAL.

File: proto/system_config.proto [hunk 2]
```diff
@@ -183,11 +183,17 @@ message SwitchSetup {
   // Local port of second DAQ OpenFlow controller
   int32 lo_port_2 = 15;
 
-  // Local port for an alternate OpenFlow controller
+  // Port for an alternate stack setup, not controlled by daq.
   int32 alt_port = 16;
 
+  // Port for alternate setup varz server.
+  int32 varz_port = 17;
```
Comment (didovesei) [inline proto/system_config.proto:190]: Would `alt_varz_port` be better? I had thought this was to configure the prometheus port for faucet/gauge before I read the comment above...

**Review by didovesei — APPROVED** [review]

**Review by henry54809 — APPROVED** [review]

File: proto/system_config.proto (hunk 2 above)
Comment (grafnu) [reply proto/system_config.proto:190]: Good catch. I renamed this to alt_varz_port and the previous one to alt_of_port

**Pull request merged**